src/ui/public/vis/vis.js

```diff
@@ -100,8 +104,9 @@ export function VisProvider(Private, indexPatterns, getAppState) {
           },
           addFilter: (data, columnIndex, rowIndex, cellValue) => {
             const agg = data.columns[columnIndex].aggConfig;
+            const columnId = data.columns[columnIndex].id;
```

Comment:
ehm, it would need to be `const { aggConfig, id } = ...` and i don't think id is a good name outside of the column context, thats why i didn't use destructuring. 

id/index ... yeah we can go with id ... or should we go with id and fallback to index (if its a number and id was not found ? )